doc/srv6/srv6_static_config_hld.md

```diff
@@ -0,0 +1,174 @@
+Then, we design a new SRv6 Manager module in bgpcfgd to subscribe to the **SRV6_MY_SID_TABLE** table and compile changes in CONFIG_DB to changes in the configurations of FRR.
+To verify the correctness of the aforementioned flow, we also define the relevant YANG model specification.
+The workflow of the new mechanism is shown in the following diagram.
+
+![Static SRv6 Config flow](images/SRv6_bgpcfgd.png)
+
+The design details of each step is described in the following subsections.
+
+## 3.1 New Table in ConfigDB
+
+
+**SRV6_MY_SID_TABLE**
+
+Description: New table to hold local SID definition and SID to behavior mapping. (A simplified redefinition of SRV6_MY_SID_TABLE in [SRv6_HLD](./srv6_hld.md))
+
+Schema:
+
+```
+; New table
+; holds local SID to behavior mapping, allow 1:1 or n:1 mapping
+
+key = SRV6_MY_SID_TABLE|ipv6address
+; field = value
+block_len = blen             ; bit length of block portion in address, default 32
+node_len = nlen              ; bit length of node ID portion in address, default 16
+func_len = flen              ; bit length of function portion in address, default 16
+arg_len = alen               ; bit length of argument portion in address, default 0
+action = behavior            ; behaviors defined for the SID
+vrf = VRF_TABLE.key          ; Optional, VRF name for decapsulation actions
+adj = address,               ; Optional, list of adjacencies for cross-connect actions
+
+For example:
+    "SRV6_MY_SID_TABLE" : {
+        "FCBB:BBBB:20::" : {
+           "action": "uN"
+        },
+        "FCBB:BBBB:20:F1::" : {
+           "action": "uDT46",
+           "vrf":  "VRF-1001"
+        },
+        "FCBB:BBBB:20:F2::" : {
+           "action": "uDT46",
+           "vrf":  "VRF-1001"
+        },
+        "FCBB:BBBB:20:F3::" : {
+           "action": "end.x",
+           "adj": [
+                FCBB:BBBB:10::1, 
+                FCBB:BBBB:10::2
+            ],
+        }
+    }
+```
+
+
+## 3.2 Bgpcfgd changes
+
+To enable automatic programming SRv6 configurations from CONFIG_DB to FRR, we need to add a new module in bgpcfgd to watch changes in **SRV6_MY_SID_TABLE** and compile the corresponding changes in FRR's configurations.
+Following the naming convention of modules in bgpcfgd, we call this new module SRv6 Manager.
+The new SRv6 Manager are supposed to verify the validity of the configuration entries coming from the CONFIG_DB.
+If it gets an invalid configuration input, it should log the event in the syslog and not compile the configuration into FRR.
+
+## 3.3 YANG Model
+The simplified version of the YANG model is defined below.
+```
+module: sonic-srv6
+  +--rw sonic-srv6
+     +--rw SRV6_MY_SID
+     |  +--rw SRV6_MY_SID_LIST* [ip-address]
+     |     +--rw ip-address    inet:ipv6-address
+     |     +--rw block_len?    uint8
+     |     +--rw node_len?     uint8
+     |     +--rw func_len?     uint8
+     |     +--rw arg_len?      uint8
+     |     +--rw action?       string
```

Comment:
we can define explicit string enum here ?  for our current usecase we can just have enum of uN/A, uDT46 ?

Reply:
changed, uA needs to be removed per your comment.